src/components/forms/AttributeForm.jsx

```diff
@@ -0,0 +1,182 @@
+            alignContent="center"
+            wrap="nowrap"
+            spacing={1}
+          >
+            {props.readOnly || props.lockId ? (
+              <Grid item>
+                <Typography variant="h6" display="inline">
+                  idField:
+                </Typography>
+                <Typography variant="h6" display="inline" color="textSecondary">
+                  &nbsp;{attributes[idAttrName]}
+                </Typography>
+              </Grid>
+            ) : (
+              <Grid item>
+                <StringField
+                  label={idAttrName}
+                  value={attributes[idAttrName]}
+                  onChange={handleSetValue}
+                  error={props.errMsg[idAttrName] ? true : false}
+                  helperText={props.errMsg[idAttrName]}
+                />
+              </Grid>
+            )}
+            {/*Key icon*/}
+            <Grid item>
+              <Tooltip title={t('modelPanels.internalId', 'Unique Identifier')}>
+                <Key
+                  fontSize="small"
+                  color="disabled"
+                  style={{ marginTop: 8 }}
+                />
+              </Tooltip>
+            </Grid>
+          </Grid>
+        </CardContent>
+
+        {Object.keys(props.dataModel).flatMap((attrName) => {
+          const attrType = props.dataModel[attrName].type;
+          const InputField = field[attrType];
+          let otherProp = {};
+          const clearButton = (
+            <InputAdornment position="end">
+              <Tooltip title="Reset">
+                <IconButton
+                  onClick={() => {
+                    handleClearButton(attrName);
+                  }}
+                  onMouseDown={(event) => event.preventDefault()}
+                >
+                  <ClearIcon />
+                </IconButton>
+              </Tooltip>
+            </InputAdornment>
+          );
+
+          if (['Integer', 'Float', 'String'].includes(attrType)) {
+            otherProp.InputProps = {
+              readOnly: props.readOnly ? true : false,
+              endAdornment: clearButton,
+            };
+          } else if (['Boolean', 'DateTime'].includes(attrType)) {
+            otherProp.disabled = props.readOnly ? true : false;
+          }
+          return props.dataModel[attrName].foreignKey ||
+            props.dataModel[attrName].isId
+            ? []
+            : [
+                <CardContent key={attrName} className={classes.cardContent}>
+                  <InputField
+                    {...otherProp}
+                    id={attrName + '-' + attrType + 'Field'}
+                    label={attrName}
+                    value={attributes[attrName]}
+                    onChange={handleSetValue}
+                    error={props.errMsg[attrName] ? true : false}
+                    helperText={props.errMsg[attrName]}
+                  />
+                </CardContent>,
+              ];
+        })}
```

Comment:
The foreign-key concept is a bit too specific for a more or less generic attribute/input form. If we don't want to render specific fields that come from the server, my suggestion would be to just pass the data that we want to be rendered.

The parent page, which sends the requests to the server, should be able to filter out attributes as needed.